### PR TITLE
Migrate to shared workflows

### PR DESCRIPTION
**Pull request by eatyourgreens**
Use Zooniverse shared workflows for deploys.

### CODE REVIEW

File: .github/workflows/deploy_production.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  build_and_push_image:
+      name: Build and Push Image
+      uses: zooniverse/ci-cd/.github/workflows/build_and_push_image.yaml@main
+      needs: get_date
+      with:
+        repo_name: subject-set-search-api
+        commit_id: ${{ github.sha }}
+        latest: true
+        build_args: |
+          BUILD_DATE: ${{ needs.get_date.outputs.date }}
```
Comment (eatyourgreens) [inline .github/workflows/deploy_production.yml:27]: `BUILD_DATE` is printed as an empty string when this runs, which means that new data isn't read from the Panoptes API.
https://github.com/zooniverse/subject-set-search-api/actions/runs/4803174012/jobs/8547411836#step:7:151

Comment (eatyourgreens) [reply .github/workflows/deploy_production.yml:27]: No wait, `BUILD_DATE` isn't printed for a successful build either. I'm not sure why this branch is ignoring it in the image.
https://github.com/zooniverse/subject-set-search-api/actions/runs/4806214177/jobs/8553461972#step:4:314

**Comment by eatyourgreens** [comment]
Builds on this branch aren't reading fresh data from Panoptes. This run deploys a cached image, rather than building a new image.
https://github.com/zooniverse/subject-set-search-api/actions/runs/4807084405/jobs/8555393785

**Comment by eatyourgreens** [comment]
https://github.com/zooniverse/subject-set-search-api/pull/25 has the same problem, which isolates it to the `cache-from` and `cache-to` parameters that are used by the shared workflow.

**Comment by eatyourgreens** [comment]
Caching might be fixed by #26, which uses a separate image to build the data CSVs and is now rebased into this branch.

**Comment by eatyourgreens** [comment]
I'll run a couple of builds on this branch and see if we get a new image each time.
- First build: https://github.com/zooniverse/subject-set-search-api/actions/runs/4810814110

**Comment by zwolf** [comment]
Looks like you rewrote history on the this repo and removed my commits from yesterday? It’s confusing to come back to in this state, so I'll leave you to it.

**Comment by eatyourgreens** [comment]
Yesterday’s commits broke production, and the lita command, so I rolled `main` back to the last working commit, then moved the changes into this branch. 

The `deploys` channel on Slack has details.

**Comment by eatyourgreens** [comment]
Builds are still cached on this branch, even after #26. ☹️

**Comment by eatyourgreens** [comment]
I’m closing this because the caching in the shared workflow breaks the data fetching step in the Docker build. 

Right now, this repo can’t use the shared ‘build and push’ workflow.

**Comment by eatyourgreens** [comment]
It might be worth revisiting this now that the build args have been fixed, in #31, and `BUILD_DATE` is baked into the Docker build. That might force a cache miss for the data fetching step.

**Pull request closed without merging**